### ISSUE TITLE
Untranslation of some words from an external dictionary

### ISSUE THREAD

**Issue by Dmitry-Sn** [huggingface/transformers#9025]
I use some pre-trained translator models from your library (for example, Helsinki-NLP). During the translation process, I would like to leave some words untranslatable (for example, acronyms, toponyms or names) due to the presence of errors in their translation.  I tried adding extra tokens and replacing these words with conditional tokens (<extra_id_0>), but this translation requires raising the num_beams parameter, which significantly slows down the translation. Unfortunately, I can't find any additional mechanisms to perform this task. Is it provided or creating it is a custom task?

**Comment by patrickvonplaten** [comment]
Hey @Dmitry-Sn,

It would be great if you could ask these kind of questions on the forum: https://discuss.huggingface.co/ . We try to keep github for issues and less for user-specific use cases. Thanks!

**Comment by vpegasus** [comment]
> 

Hi,  have you found some way to solve this problem? I meet the same situation, some proper nouns can' t translated properly, and I want to keep it as its native format.

**Comment by Dmitry-Sn** [comment]
Hi @vpegasus! There were not great ideas, generally.
As far as I remember, I decided to try to finetune the model for a special token. In my task, it was simple - since I wanted to keep geographical names in the original language, I replaced only them in the training data with a special token (I didn't have time to check the effectiveness, since I left that company).
Colleagues also suggested a solution with special characters (it seems to separate the word with <> characters), but it worked poorly.
All this applies to the models on MarianMT.
